hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/balancer/FindTargetGreedy.java

```diff
@@ -109,6 +122,7 @@ private void setConfiguration(ContainerBalancerConfiguration conf) {
   @Override
   public ContainerMoveSelection findTargetForContainerMove(
       DatanodeDetails source, Set<ContainerID> candidateContainers) {
+    sortTargetForSource(source);
```

Comment:
Sorting would happen every time function is called. I think we can optimise for same source.

Reply:
thanks, will do this optimization

Reply:
after a deep thought,  i think we have to sort potentialTargets every time , even for same source. after a certain target is selected and a move option is scheduled to it, `sizeEntering` of it will increase, and thus the utilization will increase. so when choosing a target for even the same source, if two candidate target has the same network topology distance to the source , the priority may change according to current usageinfo.

Reply:
We can remove the target and re-add it I think. But we can do this in a separate PR.